node/src/blockchain/test_utils.rs

```diff
@@ -76,73 +213,55 @@ pub struct BlockchainInterfaceMock {
     send_batch_of_payables_results:
         RefCell<Vec<Result<Vec<ProcessedPayableFallible>, PayableTransactionError>>>,
     get_transaction_receipt_params: Arc<Mutex<Vec<H256>>>,
-    get_transaction_receipt_results: RefCell<Vec<ResultForReceipt>>,
+    get_transaction_receipt_results:
+        RefCell<Vec<Result<Option<TransactionReceipt>, BlockchainError>>>,
     lower_interface_result: Option<Box<LowBlockchainIntMock>>,
     arbitrary_id_stamp_opt: Option<ArbitraryIdStamp>,
+    get_chain_results: RefCell<Vec<Chain>>,
```

Comment:
Suggestion: Would make sense to have this as `Option<Chain>` instead, because it's a value that's not believed to be ever changing over time. If you get a certain chain once you will want to get it every time. On the other hand, by that, you lose the chance of a panic because of unexpected count of calls on this method (if you prepared two results in the queue but the exercised code comes across this method three times. Which is supposed to be treated by the params assertions anyway, but I understand that it's already a lot that you'd have to bear in mind writing a good test.) That's why it's a suggestion. I'm not writing this to push you.